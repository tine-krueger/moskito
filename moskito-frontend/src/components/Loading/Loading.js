import styled, { keyframes } from 'styled-components/macro'

export default function Loading() {
    return(
        <Spinner data-testid={'loading-spinner'}/>
    )
}

const spinn = keyframes`
    0% {
        transform: rotate(0deg);
    }
    100% {
        transform: rotate(360deg);
    }
`

const Spinner = styled.div`
    overflow: hidden;
    font-size: 10px;
    margin: 50px auto;
    text-indent: -9999em;
    width: 11em;
    height: 11em;
    border-radius: 50%;
    background: var(--link);
    background: linear-gradient(to right, var(--link) 10%, rgba(255, 255, 255, 0) 42%);
    position: relative;
    animation: ${spinn} 1.4s infinite linear;
    transform: translateZ(0);

    :before {
        width: 50%;
        height: 50%;
        background: var(--link);
        border-radius: 100% 0 0 0;
        position: absolute;
        top: 0;
        left: 0;
        content: '';
    }

    :after {
        background: var(--headline);
        width: 75%;
        height: 75%;
        border-radius: 50%;
        content: '';
        margin: auto;
        position: absolute;
        top: 0;
        left: 0;
        bottom: 0;
        right: 0;
        opacity: 0.5;
    }
`