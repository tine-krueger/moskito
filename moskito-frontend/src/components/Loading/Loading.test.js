import { render } from '@testing-library/react'
import 'jest-styled-components'
import Loading from './Loading'

describe('Loading', () => {
  it('renders correctly', () => {
    const { container } = render(<Loading/>)
    expect(container.firstChild).toMatchSnapshot()
  })
})