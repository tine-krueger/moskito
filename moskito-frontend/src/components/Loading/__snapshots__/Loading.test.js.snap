// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders correctly 1`] = `
.c0 {
  overflow: hidden;
  font-size: 10px;
  margin: 50px auto;
  text-indent: -9999em;
  width: 11em;
  height: 11em;
  border-radius: 50%;
  background: var(--link);
  background: linear-gradient(to right,var(--link) 10%,rgba(255,255,255,0) 42%);
  position: relative;
  -webkit-animation: Hqxfx 1.4s infinite linear;
  animation: Hqxfx 1.4s infinite linear;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
}

.c0:before {
  width: 50%;
  height: 50%;
  background: var(--link);
  border-radius: 100% 0 0 0;
  position: absolute;
  top: 0;
  left: 0;
  content: '';
}

.c0:after {
  background: var(--headline);
  width: 75%;
  height: 75%;
  border-radius: 50%;
  content: '';
  margin: auto;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  opacity: 0.5;
}

<div
  class="c0"
/>
`;
